w3ctrace/state.go

```diff
@@ -11,24 +11,56 @@ import (
 // VendorInstana is the Instana vendor key in the `tracestate` list
 const VendorInstana = "in"
 
+// Max amount of KV pairs in `tracestate` header
+const maxKVPairs = 32
+
+// Length of entries that should be filtered first in case, if tracestate has more than `maxKVPairs` items
+const thresholdLen = 128
+
 // State is list of key=value pairs representing vendor-specific data in the trace context
 type State []string
 
-// ParseState parses the value of `tracestate` header. It strips any optional white-space chararacters
-// preceding or following the key=value pairs. Empty list items are omitted.
+// ParseState parses the value of `tracestate` header. Empty list items are omitted.
 func ParseState(traceStateValue string) (State, error) {
 	var state State
 
-	for _, st := range strings.SplitN(traceStateValue, ",", 32) {
-		st = strings.TrimSpace(st)
-		if st == "" {
-			continue
+	entries := strings.Split(traceStateValue, ",")
+	if len(entries) == 1 && entries[0] == "" {
+		return state, nil
+	}
```

Comment:
This looks like a premature optimization: the `range` statement inside `filterEmptyItems` will not be executed for an empty list, so the check at L35 will suffice.

Reply:
This is changed now 👍